src/components/ui/number-input.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { cn } from '@/lib/utils';
+
+export interface NumberInputProps
+  extends React.InputHTMLAttributes<HTMLInputElement> {
+  hideSpinButtons?: boolean;
+  decimalPlaces?: number;
+}
+
+const NumberInput = React.forwardRef<HTMLInputElement, NumberInputProps>(
+  (
+    {
+      className,
+      type = 'number',
+      hideSpinButtons = false,
+      decimalPlaces,
+      value,
+      ...props
+    },
+    ref
+  ) => {
+    let formattedValue = value;
+
+    if (decimalPlaces !== undefined && value !== undefined && value !== '') {
```

Comment:
Please remove this part. As mentioned [here](https://github.com/grants-stack-frontier/op-rf4-voting-ui/issues/181#issuecomment-2369897399), fixing the decimal places to 2 should be handled by the API, not the frontend. I am unable to edit the allocation to the amount I want and when I type numbers, I see the OP amount changing but the input amount does not.

Reply:
removed. i'm still getting 10 decimal places from the backend though